### ISSUE TITLE
Leveraging Docker for the build/test process?

### ISSUE THREAD

**Issue by zmoazeni** [github/gh-ost#740]
This is a meatier issue, but I'm interested in simplifying some of the build/cibuild scripts by using [the golang-cross docker image](https://hub.docker.com/r/dockercore/golang-cross) instead of downloading/verifying the actual Go version for the host environment.

Some other areas of simplification I can see:

* Drop the `.gopath`. This would be part of the mapped docker volume instead.
* Possibly bundling the binary of dbdeployer inside the image which might affect `gh-ost-ci-env`. But a side effect of this could be possibly parallelizing the MySQL version localtests.

It'll take a bit of effort and experimentation, but I wanted to reach out to see if this was interesting before spending I began spending the effort. We should be [a-okay on the Travis-CI's side](https://docs.travis-ci.com/user/docker/).

My goal here would be to simplify bootstrapping and testing the codebase. Both with the script maintenance and the dev-groking.

**Comment by shlomi-noach** [comment]
@zmoazeni I'm on board with this. You may have noticed I'm a bit slow to respond, but otherwise this should be good.

**Comment by zmoazeni** [comment]
> You may have noticed I'm a bit slow to respond, but otherwise this should be good.

No worries. I just figured you're busy on other things at the moment. I'll put something together and see if you like it.

**Comment by zmoazeni** [comment]
@shlomi-noach I have something working locally for build/fmt/testing. I wanted to ask though, is `./script/cibuild-gh-ost-build-deploy-tarball` used? I'm wondering if I need to convert that or if I can just drop it.

**Comment by shlomi-noach** [comment]
@zmoazeni yeah, `./script/cibuild-gh-ost-build-deploy-tarball` is used internally at GitHub and is the build script used by our CI/CD to deploy `gh-ost`.
